codersdk/name_test.go

```diff
@@ -254,3 +256,40 @@ func TestUserRealNameValid(t *testing.T) {
 		})
 	}
 }
+
+func TestGroupNameValid(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		Name  string
+		Valid bool
+	}{
+		{"", false},
+		{"my-group", true},
+		{"create", false},
+		{"new", false},
```

Comment:
Why is this invalid?

Reply:
To avoid route conflicts. 

```golang
// Avoid conflicts with routes like /templates/new and /groups/create.
if str == "new" || str == "create" {
	return xerrors.Errorf("cannot use %q as a name", str)
}
```